pinot-plugins/pinot-file-system/pinot-gcs/src/main/java/org/apache/pinot/plugin/filesystem/GcsPinotFS.java

```diff
@@ -204,7 +205,7 @@ public boolean delete(URI segmentUri, boolean forceDelete) throws IOException {
     LOGGER.info("Deleting uri {} force {}", segmentUri, forceDelete);
     try {
       if (!exists(segmentUri)) {
-        return false;
+        return forceDelete;
       }
       if (isDirectory(segmentUri)) {
         if (!forceDelete) {
```

Comment:
If directory is not empty, it should return `false` instead of throwing exception